src/webapi/selection.rs

```diff
@@ -0,0 +1,341 @@
+        js! (
+            return @{self}.focusNode;
+        ).try_into().unwrap()
+    }
+
+    /// Returns the number of characters that the selection's anchor is offset within the
+    /// [focus_node](struct.Selection.html#method.focus_node).
+    ///
+    /// This number is zero-based. If the selection begins with the first character in the
+    /// [focus_node](struct.Selection.html#method.focus_node), 0 is returned.
+    ///
+    /// [(Javascript docs)](https://developer.mozilla.org/en-US/docs/Web/API/Selection/focusOffset)
+    pub fn focus_offset(&self) -> u32 {
+        js! (
+            return @{self}.focusOffset;
+        ).try_into().unwrap()
+    }
+
+    /// Returns a boolean which indicates whether or not there is currently any text selected; That
+    /// is to say that the selection's start and end points are at the same position in the
+    /// content.
+    ///
+    /// Keep in mind that a collapsed selection may still have one (or more, in Gecko)
+    /// [Range](struct.Range.html)s, so [range_count](struct.Selection.html#method.range_count) may
+    /// not be zero. In that scenario, calling a [Selection](struct.Selection.html) object's
+    /// [get_range_at](struct.Selection.html#method.get_range_at) method may return a
+    /// [Range](struct.Range.html) object which is collapsed.
+    ///
+    /// [(Javascript docs)](https://developer.mozilla.org/en-US/docs/Web/API/Selection/isCollapsed)
+    pub fn is_collapsed(&self) -> bool {
+        js! (
+            return @{self}.isCollapsed;
+        ).try_into().unwrap()
+    }
+
+    /// Returns the number of ranges in the selection.
+    ///
+    /// Before the user has clicked a freshly loaded page, the
+    /// [range_count](struct.Selection.html#method.range_count) is 0. After the user
+    /// clicks on the page, [range_count](struct.Selection.html#method.range_count) even if no 
+    /// selection is visible.
+    ///
+    /// A user can normally only select one range at a time, so the 
+    /// [range_count](struct.Selection.html#method.range_count) will usually be 1.
+    /// Scripting can be used to make the selection contain more than 1 range.
+    ///
+    /// Gecko browsers allow multiple selections across table cells.
+    ///
+    /// [(Javascript docs)](https://developer.mozilla.org/en-US/docs/Web/API/Selection/rangeCount)
+    pub fn range_count(&self) -> u32 {
+        js! (
+            return @{self}.rangeCount;
+        ).try_into().unwrap()
+    }
+
+    /// Returns the type of the current selection.
+    ///
+    /// Possible values are:
+    ///
+    /// * `None`: No selection has currently been made.
+    /// * `Caret`: The selection is collapsed (i.e. the caret is placed on some text, but no
+    /// range has been selected).
+    /// * `Range`: A range has been selected.
+    ///
+    /// [(Javascript docs)](https://developer.mozilla.org/en-US/docs/Web/API/Selection/type)
+    pub fn selection_type(&self) -> String {
```

Comment:
Since as far as I can see it can only return three different strings this should return an enum.